app/models/id_card_attributes.rb

```diff
@@ -60,15 +60,18 @@ def branches_of_service
   end
 
   def discharge_types
-    discharges = if Settings.vsp_environment == 'production' || !Flipper.enabled?(:military_information_vaprofile)
+    discharges = if !Flipper.enabled?(:military_information_vaprofile_vic)
                    @user.military_information
                         .service_episodes_by_date
                         .map(&:discharge_character_of_service_code)
-                 elsif Flipper.enabled?(:military_information_vaprofile)
-                   military_info
-                     .service_episodes_by_date
-                     .map(&:character_of_discharge_code)
+                 elsif Flipper.enabled?(:military_information_vaprofile_vic)
+                   ## If the discharge code is one of the known, unwanted three-character
+                   ##  codes from VA Profile, replace it with nil.
+                   invalid_codes = %w[DVN DVU CVI VNA]
```

Comment:
Do we know how often VIC changes codes (if at all)? I know the VIC folks had mentioned any code > 3 characters should be returned as nil. Instead of hard coding the values, maybe we could check if the code > 3 characters and return nil. I'm also fine with us rolling this out as rev 1 and implementing the > 3 character logic as the next rev

Reply:
@LindseySaari Lisa said in her email yesterday:
> We do not want a broad rule, just one that applies to the 4 specific multi-character one’s were aware of at this time.  This will allow us to catch any future anomalies. 

So it seems like they _want_ to do it this way.